File: Examples/UI/WinForm/VB.NET/Scrollbars/NHScrollBarUC.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Diagnostics
Imports System.Collections
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Runtime.InteropServices
Imports System.Windows.Forms

Imports Nevron.UI.WinForm.Controls
Imports Nevron.Examples.Framework.WinForm

Namespace Nevron.Examples.UI.WinForm
	Public Class NHScrollBarUC
		Inherits NExampleUserControl
		#Region "Constructor"

		Public Sub New(ByVal f As MainForm)
			MyBase.New(f)
			InitializeComponent()

			Dock = DockStyle.Fill
		End Sub


		#End Region

		#Region "Implementation"

		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not components Is Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub


		#End Region

		#Region "Event Handlers"

		Private Sub nhScrollBar1_ValueChanged(ByVal sender As Object, ByVal e As Nevron.UI.WinForm.Controls.ScrollBarEventArgs) Handles nhScrollBar1.ValueChanged
		End Sub

		Private Sub nCheckBox1_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles nCheckBox1.CheckedChanged
			For Each c As Control In Controls
				If TypeOf c Is NScrollBar Then
					c.Enabled = nCheckBox1.Checked
				End If
			Next c
		End Sub

		Private Sub nCheckBox2_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles nCheckBox2.CheckedChanged
			For Each c As Control In Controls
				If TypeOf c Is NScrollBar Then
					CType(c, NScrollBar).HotTrack = Me.nCheckBox2.Checked
				End If
			Next c
		End Sub


		#End Region

		#Region "Component Designer generated code"
		''' <summary> 
		''' Required method for Designer support - do not modify 
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
			Me.nhScrollBar4 = New Nevron.UI.WinForm.Controls.NHScrollBar()
			Me.nhScrollBar3 = New Nevron.UI.WinForm.Controls.NHScrollBar()
			Me.nhScrollBar2 = New Nevron.UI.WinForm.Controls.NHScrollBar()
			Me.nhScrollBar1 = New Nevron.UI.WinForm.Controls.NHScrollBar()
			Me.nCheckBox1 = New Nevron.UI.WinForm.Controls.NCheckBox()
			Me.nCheckBox2 = New Nevron.UI.WinForm.Controls.NCheckBox()
			Me.SuspendLayout()
			' 
			' nhScrollBar4
			' 
			Me.nhScrollBar4.Anchor = (CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles))
			Me.nhScrollBar4.Location = New System.Drawing.Point(8, 72)
			Me.nhScrollBar4.Name = "nhScrollBar4"
			Me.nhScrollBar4.Size = New System.Drawing.Size(408, 56)
			Me.nhScrollBar4.TabIndex = 3
			Me.nhScrollBar4.Text = "nhScrollBar4"
			' 
			' nhScrollBar3
			' 
			Me.nhScrollBar3.Anchor = (CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles))
			Me.nhScrollBar3.Location = New System.Drawing.Point(8, 136)
			Me.nhScrollBar3.Name = "nhScrollBar3"
			Me.nhScrollBar3.Size = New System.Drawing.Size(408, 136)
			Me.nhScrollBar3.TabIndex = 2
			Me.nhScrollBar3.Text = "nhScrollBar3"
			' 
			' nhScrollBar2
			' 
			Me.nhScrollBar2.Anchor = (CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles))
			Me.nhScrollBar2.Location = New System.Drawing.Point(8, 32)
			Me.nhScrollBar2.Name = "nhScrollBar2"
			Me.nhScrollBar2.Size = New System.Drawing.Size(408, 32)
			Me.nhScrollBar2.TabIndex = 1
			Me.nhScrollBar2.Text = "nhScrollBar2"
			' 
			' nhScrollBar1
			' 
			Me.nhScrollBar1.Anchor = (CType(((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Left) Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles))
			Me.nhScrollBar1.LargeChange = 5
			Me.nhScrollBar1.Location = New System.Drawing.Point(8, 5)
			Me.nhScrollBar1.Maximum = 255
			Me.nhScrollBar1.Name = "nhScrollBar1"
			Me.nhScrollBar1.Size = New System.Drawing.Size(408, 17)
			Me.nhScrollBar1.TabIndex = 0
			Me.nhScrollBar1.Text = "nhScrollBar1"
'			Me.nhScrollBar1.ValueChanged += New Nevron.UI.WinForm.Controls.ScrollBarEventHandler(Me.nhScrollBar1_ValueChanged);
			' 
			' nCheckBox1
			' 
			Me.nCheckBox1.Checked = True
			Me.nCheckBox1.CheckState = System.Windows.Forms.CheckState.Checked
			Me.nCheckBox1.Location = New System.Drawing.Point(8, 280)
			Me.nCheckBox1.Name = "nCheckBox1"
			Me.nCheckBox1.Size = New System.Drawing.Size(80, 24)
			Me.nCheckBox1.TabIndex = 4
			Me.nCheckBox1.Text = "&Enable"
'			Me.nCheckBox1.CheckedChanged += New System.EventHandler(Me.nCheckBox1_CheckedChanged);
			' 
			' nCheckBox2
			' 
			Me.nCheckBox2.Checked = True
			Me.nCheckBox2.CheckState = System.Windows.Forms.CheckState.Checked
			Me.nCheckBox2.Location = New System.Drawing.Point(96, 280)
			Me.nCheckBox2.Name = "nCheckBox2"
			Me.nCheckBox2.Size = New System.Drawing.Size(88, 24)
			Me.nCheckBox2.TabIndex = 5
			Me.nCheckBox2.Text = "&Hot Track"
'			Me.nCheckBox2.CheckedChanged += New System.EventHandler(Me.nCheckBox2_CheckedChanged);
			' 
			' NHScrollBarUC
			' 
			Me.Controls.Add(Me.nCheckBox2)
			Me.Controls.Add(Me.nCheckBox1)
			Me.Controls.Add(Me.nhScrollBar4)
			Me.Controls.Add(Me.nhScrollBar3)
			Me.Controls.Add(Me.nhScrollBar2)
			Me.Controls.Add(Me.nhScrollBar1)
			Me.DockPadding.All = 5
			Me.Name = "NHScrollBarUC"
			Me.Size = New System.Drawing.Size(424, 312)
			Me.ResumeLayout(False)

		End Sub
		#End Region

		#Region "Fields"

		Private components As System.ComponentModel.Container = Nothing
		Private WithEvents nhScrollBar1 As Nevron.UI.WinForm.Controls.NHScrollBar
		Private nhScrollBar2 As Nevron.UI.WinForm.Controls.NHScrollBar
		Private nhScrollBar3 As Nevron.UI.WinForm.Controls.NHScrollBar
		Private WithEvents nCheckBox1 As Nevron.UI.WinForm.Controls.NCheckBox
		Private WithEvents nCheckBox2 As Nevron.UI.WinForm.Controls.NCheckBox
		Private nhScrollBar4 As Nevron.UI.WinForm.Controls.NHScrollBar

		#End Region
	End Class
End Namespace

File: Examples/UI/WinForm/CSharp/Scrollbars/NHScrollBarUC.cs
using System;
using System.Diagnostics;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Runtime.InteropServices;
using System.Windows.Forms;

using Nevron.UI.WinForm.Controls;
using Nevron.Examples.Framework.WinForm;

namespace Nevron.Examples.UI.WinForm
{
	public class NHScrollBarUC : NExampleUserControl
	{
		#region Constructor

		public NHScrollBarUC(MainForm f) : base(f)
		{
			InitializeComponent();

			Dock = DockStyle.Fill;
		}


		#endregion

		#region Implementation

		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}


		#endregion

		#region Event Handlers

		private void nhScrollBar1_ValueChanged(object sender, Nevron.UI.WinForm.Controls.ScrollBarEventArgs e)
		{
		}

		private void nCheckBox1_CheckedChanged(object sender, System.EventArgs e)
		{
			foreach(Control c in Controls)
			{
				if(c is NScrollBar)
					c.Enabled = nCheckBox1.Checked;
			}
		}

		private void nCheckBox2_CheckedChanged(object sender, System.EventArgs e)
		{
			foreach(Control c in Controls)
			{
				if(c is NScrollBar)
					((NScrollBar)c).HotTrack = this.nCheckBox2.Checked;
			}
		}


		#endregion

		#region Component Designer generated code
		/// <summary> 
		/// Required method for Designer support - do not modify 
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.nhScrollBar4 = new Nevron.UI.WinForm.Controls.NHScrollBar();
			this.nhScrollBar3 = new Nevron.UI.WinForm.Controls.NHScrollBar();
			this.nhScrollBar2 = new Nevron.UI.WinForm.Controls.NHScrollBar();
			this.nhScrollBar1 = new Nevron.UI.WinForm.Controls.NHScrollBar();
			this.nCheckBox1 = new Nevron.UI.WinForm.Controls.NCheckBox();
			this.nCheckBox2 = new Nevron.UI.WinForm.Controls.NCheckBox();
			this.SuspendLayout();
			// 
			// nhScrollBar4
			// 
			this.nhScrollBar4.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right)));
			this.nhScrollBar4.Location = new System.Drawing.Point(8, 72);
			this.nhScrollBar4.Name = "nhScrollBar4";
			this.nhScrollBar4.Size = new System.Drawing.Size(408, 56);
			this.nhScrollBar4.TabIndex = 3;
			this.nhScrollBar4.Text = "nhScrollBar4";
			// 
			// nhScrollBar3
			// 
			this.nhScrollBar3.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right)));
			this.nhScrollBar3.Location = new System.Drawing.Point(8, 136);
			this.nhScrollBar3.Name = "nhScrollBar3";
			this.nhScrollBar3.Size = new System.Drawing.Size(408, 136);
			this.nhScrollBar3.TabIndex = 2;
			this.nhScrollBar3.Text = "nhScrollBar3";
			// 
			// nhScrollBar2
			// 
			this.nhScrollBar2.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right)));
			this.nhScrollBar2.Location = new System.Drawing.Point(8, 32);
			this.nhScrollBar2.Name = "nhScrollBar2";
			this.nhScrollBar2.Size = new System.Drawing.Size(408, 32);
			this.nhScrollBar2.TabIndex = 1;
			this.nhScrollBar2.Text = "nhScrollBar2";
			// 
			// nhScrollBar1
			// 
			this.nhScrollBar1.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right)));
			this.nhScrollBar1.LargeChange = 5;
			this.nhScrollBar1.Location = new System.Drawing.Point(8, 5);
			this.nhScrollBar1.Maximum = 255;
			this.nhScrollBar1.Name = "nhScrollBar1";
			this.nhScrollBar1.Size = new System.Drawing.Size(408, 17);
			this.nhScrollBar1.TabIndex = 0;
			this.nhScrollBar1.Text = "nhScrollBar1";
			this.nhScrollBar1.ValueChanged += new Nevron.UI.WinForm.Controls.ScrollBarEventHandler(this.nhScrollBar1_ValueChanged);
			// 
			// nCheckBox1
			// 
			this.nCheckBox1.Checked = true;
			this.nCheckBox1.CheckState = System.Windows.Forms.CheckState.Checked;
			this.nCheckBox1.Location = new System.Drawing.Point(8, 280);
			this.nCheckBox1.Name = "nCheckBox1";
			this.nCheckBox1.Size = new System.Drawing.Size(80, 24);
			this.nCheckBox1.TabIndex = 4;
			this.nCheckBox1.Text = "&Enable";
			this.nCheckBox1.CheckedChanged += new System.EventHandler(this.nCheckBox1_CheckedChanged);
			// 
			// nCheckBox2
			// 
			this.nCheckBox2.Checked = true;
			this.nCheckBox2.CheckState = System.Windows.Forms.CheckState.Checked;
			this.nCheckBox2.Location = new System.Drawing.Point(96, 280);
			this.nCheckBox2.Name = "nCheckBox2";
			this.nCheckBox2.Size = new System.Drawing.Size(88, 24);
			this.nCheckBox2.TabIndex = 5;
			this.nCheckBox2.Text = "&Hot Track";
			this.nCheckBox2.CheckedChanged += new System.EventHandler(this.nCheckBox2_CheckedChanged);
			// 
			// NHScrollBarUC
			// 
			this.Controls.Add(this.nCheckBox2);
			this.Controls.Add(this.nCheckBox1);
			this.Controls.Add(this.nhScrollBar4);
			this.Controls.Add(this.nhScrollBar3);
			this.Controls.Add(this.nhScrollBar2);
			this.Controls.Add(this.nhScrollBar1);
			this.DockPadding.All = 5;
			this.Name = "NHScrollBarUC";
			this.Size = new System.Drawing.Size(424, 312);
			this.ResumeLayout(false);

		}
		#endregion

		#region Fields

		private System.ComponentModel.Container components = null;
		private Nevron.UI.WinForm.Controls.NHScrollBar nhScrollBar1;
		private Nevron.UI.WinForm.Controls.NHScrollBar nhScrollBar2;
		private Nevron.UI.WinForm.Controls.NHScrollBar nhScrollBar3;
		private Nevron.UI.WinForm.Controls.NCheckBox nCheckBox1;
		private Nevron.UI.WinForm.Controls.NCheckBox nCheckBox2;
		private Nevron.UI.WinForm.Controls.NHScrollBar nhScrollBar4;

		#endregion
	}
}
